test/unit/inbound/process-submitted-settlement.test.js

```diff
@@ -0,0 +1,137 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+describe('process submitted settlement request', () => {
+  jest.mock('../../../app/inbound/get-existing-settlement')
+  const getExistingSettlement = require('../../../app/inbound/get-existing-settlement')
+
+  jest.mock('../../../app/inbound/save-settlement')
+  const saveSettlement = require('../../../app/inbound/save-settlement')
+
+  const processSubmittedSettlement = require('../../../app/inbound/process-submitted-settlement')
+
+  let settlement
+
+  beforeEach(() => {
+    settlement = {
+      sourceSystem: 'SITIAgri',
+      invoiceNumber: 'S123456789A123456V003',
+      frn: 1234567890,
+      currency: 'GBP',
+      value: 30000,
+      settlementDate: 'Fri Jan 21 2022 10:38:44 GMT+0000 (Greenwich Mean Time)',
+      reference: 'PY1234567',
+      settled: true
+    }
```

Comment:
could you move this into its own test... constant... file... thang

it means that any other tests that need to use a mockSettlement can import your one which has many benefits:

1. less time needing to spend creating mockSettlement objects; THE ONE already exists
2. this settlement object is very sim. to an ACTUAL settlement object which means someone looking at this can understand what info it contains -- plus it means the tests are based on 'real-life' objects which improves their quality
3. consistency, consistency, consistency: when you've used this a few times you can predict and expect the outcomes yer going to get w/out even having to look at the mock object again

Reply:
for reference I've created a file called `mock-payment-request` which a sim. object exported out of it

then when I import it into file it looks like this: `JSON.parse(JSON.stringify(require('../../mock-payment-request').processingPaymentRequest))`

the end part where I'm asking for `.processingPaymentRequest` is only because I'm exporting multiple values in tt file

Reply:
explanation was very helpful, have actioned thanks